app_flutter/lib/state/flutter_build.dart

```diff
@@ -73,69 +64,101 @@ class FlutterBuildState extends ChangeNotifier {
   static const String errorMessageFetchingBranches =
       'An error occured fetching branches from flutter/flutter on Cocoon.';
 
-  /// Start a fixed interval loop that fetches build state updates based on [refreshRate].
-  // TODO(ianh): make this automatically trigger when listeners are added, and cancel when they're removed.
-  Future<void> startFetchingUpdates() async {
-    if (refreshTimer != null && refreshTimer.isActive) {
-      // There's already an update loop, no need to make another.
-      return;
-    }
+  /// How often to query the Cocoon backend for the current build state.
+  @visibleForTesting
+  final Duration refreshRate = const Duration(seconds: 10);
 
-    /// [Timer.periodic] does not necessarily run at the start of the timer.
-    _fetchBuildStatusUpdate();
+  /// Timer that calls [_fetchStatusUpdates] on a set interval.
+  @visibleForTesting
+  @protected
+  Timer refreshTimer;
 
-    _fetchFlutterBranches().then((List<String> branchResponse) => _branches = branchResponse);
+  // There's no way to cancel futures in the standard library so instead we just track
+  // if we've been disposed, and if so, we drop everything on the floor.
+  bool _active = true;
```

Comment:
Obviously we implement this with Streams then :P

Reply:
Or some sort of cancelable API, yeah. We could. Not sure it's really worth it though. At the end of the day this boils down to a network request and it's not clear how much we win from canceling that.